### PR TITLE
12 feedback notification

### PR DESCRIPTION
**Pull request by arcus76**
Merged the functionality prototype:

1)
Generate a patient and a practitioner
Overview for all instructions, patients and practitioners
Create instructions or import MedicationStatements from HAPI FHIR server
Assign an instruction from a practitioner to a patient
Patient leaves feedback (rating + comment)
Patient receives notification, if no feedback after deadline
Practitioner receives notification, that patient left feedback
Practitioner can acknowledge feedback
Both practitioner and patient see, that feedback acknowledged

2)
Merged with current state of Merge branch:
Big refactoring of both branches

3)Implemented multiple instructions/feedbacks for patient/practitioner

### CODE REVIEW

File: src/main/java/com/wise23/chariteed/controller/TestController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package com.wise23.chariteed.controller;
```
Comment (Miao-99) [inline src/main/java/com/wise23/chariteed/controller/TestController.java:1]: remember to remove this test controller at final version

File: src/main/java/com/wise23/chariteed/model/InstructionToPatient.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    @Column(name = "second_patient_comment")
+    String secondPatientComment;
+
+    @Column(name = "second_feedback_given")
+    Boolean secondFeedbackGiven = false;
+
+    @Column(name = "second_feedback_opened")
+    Boolean secondFeedbackOpened = false;
+
+}
```
Comment (Miao-99) [inline src/main/java/com/wise23/chariteed/model/InstructionToPatient.java:77]: first and second feedback? not sure if this is a scaleable solution

Comment (Miao-99) [reply src/main/java/com/wise23/chariteed/model/InstructionToPatient.java:77]: maybe its better to implement this feedback in PatientFeedbackData

File: src/main/java/com/wise23/chariteed/repository/InstructionToPatientRepository.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+@Repository
+public interface InstructionToPatientRepository extends JpaRepository<InstructionToPatient, Long> {
+
+
+    @Query("SELECT i FROM InstructionToPatient i WHERE i.patient.fhirId = ?1 AND i.givenAt IN (SELECT MAX(i_latest.givenAt) FROM InstructionToPatient i_latest WHERE i_latest.patient.fhirId = ?1)")
+    Optional<InstructionToPatient> findLatestInstructionOfPatient(Long fhirID);
+
+    @Query("SELECT i FROM InstructionToPatient i WHERE i.patient.fhirId = ?1 AND (i.firstFeedbackGiven = FALSE OR i.secondFeedbackGiven = FALSE)")
+    List<InstructionToPatient> findInstructionsOfPatientByFeedbackGivenIsFalseOrSecondFeedbackGivenIsFalse(Long fhirID);
+}
```
Comment (Miao-99) [inline src/main/java/com/wise23/chariteed/repository/InstructionToPatientRepository.java:20]: is it for searching external instructions?

File: src/main/resources/templates/homepage.html [hunk 4]
```diff
@@ -23,5 +23,12 @@ <h2>Welcome to the Charité Emergency Department App</h2>
     <a th:href="@{/logout}">Log out</a>
   </div>
 
+  <div>
+    <a th:href="@{/test/generatesubjects}" class="btn btn-info">Generate a patient and a practitioner</a>
+    <a th:href="@{/patient/view/all}" class="btn btn-info">View all patients</a>
+    <a th:href="@{/practitioner/view/all}" class="btn btn-info">View all practitioners</a>
+    <a th:href="@{/instruction/view/all}" class="btn btn-info">View all instructions</a>
```
Comment (Miao-99) [inline src/main/resources/templates/homepage.html:30]: could also add redirect links in admin/practitioner/patient dashboards

File: src/main/resources/templates/practitioner/dashboard/error.html [hunk 5]
```diff
@@ -6,8 +6,8 @@
 </head>
 <body>
 	<h1>ERROR: Patient ID does not exist, no valid condition entry or patient already existing.</h1>
```
Comment (Miao-99) [inline src/main/resources/templates/practitioner/dashboard/error.html:8]: file name error.html could be more specific

**Review by Miao-99 — APPROVED** [review]

File: src/main/java/com/wise23/chariteed/model/InstructionToPatient.java (hunk 2 above)
Comment (arcus76) [reply src/main/java/com/wise23/chariteed/model/InstructionToPatient.java:77]: No, it is not scalable:)
Probably yes, a separate class with a separate table, with a type (1st wave feedback, 2nd wave feedback and app note)

File: src/main/java/com/wise23/chariteed/repository/InstructionToPatientRepository.java (hunk 3 above)
Comment (arcus76) [reply src/main/java/com/wise23/chariteed/repository/InstructionToPatientRepository.java:20]: Nope, local instructions.
Hapi fhir search would be somewhere in service

File: src/main/resources/templates/practitioner/dashboard/error.html (hunk 5 above)
Comment (arcus76) [reply src/main/resources/templates/practitioner/dashboard/error.html:8]: This is probably @JannesVolkens files, so I just moved them all from a "doctor" folder to "practitioner" folder

**Pull request merged**